src/mono/browser/debugger/DebuggerTestSuite/MiscTests.cs

```diff
@@ -1199,5 +1199,35 @@ public async Task TestDebugUsingMultiThreadedRuntime(int _attempt)
             Assert.Equal(locals[1]["value"]["type"], "number");
             Assert.Equal(locals[1]["name"], "currentThread");
         }
+
+        [ConditionalFact(nameof(RunningOnChrome))]
+        public async Task CheckDebuggerEvalScriptsAreNotSentToIDE()
+        {
+           var bp = await SetBreakpoint("dotnet://debugger-test.dll/debugger-test.cs", 10, 8);
+
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_add(); invoke_add(); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs", 10, 8,
+                "Math.IntAdd",
+                locals_fn: async (locals) =>
+                {
+                    CheckNumber(locals, "a", 10);
+                    await Task.CompletedTask;
+                }
+            );
+            var oldValue = noUrlScripts;
```

Comment:
This seems brittle and may break if we add something that sends `url=""` events for some other reason. Instead can we use the same check as the proxy - check the top frame function name, and ensure their count is zero?

Reply:
Please let me know if you still want me to change the test.

Reply:
So are we trying to identify the case where we might send `url=""` to VS which isn't a debugger command?

Reply:
I'm trying to identify an increase in the number of scripts with `url=""` sent to VS when we are only having 2 pauses in a breakpoint, which may cause the slowness in VS. I can also remove the `url=""` check, the scripts should not be increased in this scenario at all.

Reply:
Ok, fair enough. We can respond if we hit any issues in future which should also give us info about what these other cases are.